src/internal/m365/onedrive/restore.go

```diff
@@ -683,7 +708,12 @@ func createRestoreFolders(
 		}
 
 		// create the folder if not found
-		folderItem, err = fr.PostItemInContainer(ictx, driveID, parentFolderID, newItem(folder, true))
+		folderItem, err = fr.PostItemInContainer(
+			ictx,
+			driveID,
+			parentFolderID,
+			newItem(folder, true),
+			control.Replace) // folders can be replaced without issue.
```

Comment:
Could you add a note as well that this will help us avoid race conditions where a folder is created between the `GetFolderByName` call above and here

Reply:
Sure, can do.